src/Confluent.Kafka/Serialization/LongDeserializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+
+using System;
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    /// Derializer for Int64 data
+    /// </summary>
+    public class LongDeserializer : IDeserializer<long>
+    {
+        /// <summary>
+        /// Deserialize data to Int64
+        /// </summary>
+        /// <param name="data">input data</param>
+        /// <returns>deserialize result</returns>
+        public long Deserialize(byte[] data)
+        {
+            if (data == null)
+            {
+                // in accordance with java client.
+                return default(long);
```

Comment:
@ewencp - I actually think it might be less confusing to have the two types. But to your second point, I have to admit I was completely asleep yesterday and imagining tomb-stoning was by done by setting the key to `null` (which of course is comically wrong). Now i'm thinking straight, I have a definite bias towards not supporting `null` values in numeric serializers as they won't be as commonly used as I was errantly thinking.

@treziac and repliers - in general agreement here - the API is not necessarily the best (though i imagine the people coming up with it were probably making tradeoffs). And we shouldn't deviate.

@wangyanjun 's PR LGTM except for some comment formatting / english things which I'm happy to change myself. I'll try to do that inline on github or push to your branch, unsure what's going to work.